packages/block-library/src/buttons/style.scss

```diff
@@ -1,6 +1,7 @@
 // Increased specificity to override blocks default margin.
 .wp-block-buttons .wp-block-button.wp-block-button {
```

Comment:
The fix seems reasonable but together with this highly specific selector I think it could easily override themes? This might be better as a separate issue but I'd like to mention it since it could override themes. @jasmussen if you dont mind me pinging you

I don't actually see why this needs to be so specific. As far as I can tell `.wp-block-buttons .wp-block-button` should be fine. `.wp-block-button` is unfortunately not possible due to lists having a margin-left specified.

https://github.com/WordPress/gutenberg/commit/cee64399ba605ce65a08df3e5a2f0381b6bd3bf4#diff-de646721c39fc8240db5836e29873266R1 (@youknowriad do you happen to know if there's been a change since your commit that would allow to lower the specificity? I tested lowering it and my initial testing looked fine for twentynineteen and twentytwenty)
https://github.com/WordPress/gutenberg/commit/a3f1e6545af45904957bb7648938e33cf759938a#diff-ee2ed3adbe2578628039530c717a9a93R255 (above commit overrides this)
https://github.com/WordPress/gutenberg/commit/2ea41bf03762cd9874e066c5456181d168fddeaf#diff-b175fef433360d2d2c873fdd717e5f05R108 (above commit overrides this due to https://github.com/WordPress/gutenberg/pull/20951/files#diff-947e2eca7278f9a543620b68d7187d09)

Reply:
Always happy to receive pings! 

Yes, in principle, the less specific a selector we can make, the better. But in cases where it's necessary, then of course the button needs the least amount of specificity. If we can reduce it, including for the reasons you mention, all the better. I haven't tested, so I can't verify it's the case, but of course it's worth double checking.

At a higher level, this is an excellent example of why I created #22208. Right now, editor-only, _every block_ gets top and bottom margins. This is regardless of whether they're meant to have top and bottom margins, or not.

There's some conversation in that pull request heading in the direction of changing this baseline margin to be a CSS variable for more theme control, and even to expand it so it isn't editor only but both editor and frontend. Feel free to chime in on that PR.

Reply:
I don't really know whether we can reduce the specificity at the moment. Worth trying though in a PR

Reply:
> I don't really know whether we can reduce the specificity at the moment. Worth trying though in a PR

I'll test it some more and send a PR if I can't find any issues myself. I think the 0,3,0 specificity would most likely trump any and all theme styles while 0,2,0 would be more common in the real world.

Reply:
The overly specific selector (which wasn't technically related to this PR anyway) was fixed by #23246. :partying_face: